engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java

```diff
@@ -418,7 +430,8 @@ private CachedCurrentDate(@NotNull final ZoneId timeZone) {
 
         @Override
         void update(final long currentTimeMillis) {
-            value = formatDate(epochMillisToInstant(currentTimeMillis), timeZone);
+            date = toLocalDate(epochMillisToInstant(currentTimeMillis), timeZone);
+            str = formatDate(date);
             valueExpirationTimeMillis =
                     epochMillis(atMidnight(epochNanosToInstant(millisToNanos(currentTimeMillis) + DAY), timeZone));
```

Comment:
I know this is existing code, but it doesn't look correct to me - adding 24 hours of nanos might not account for leap occurrences. Think this needs to be done wrt `date.plusDays(1)`.